lab1/app.js

```diff
@@ -74,38 +78,89 @@ class Cart {
 
     // render a list of item under root element
     render () {
-		let tbody = this.root.querySelector('tbody');
-		//tbody.innerHTML = <tr>
-			//<td></td>
-			//<td></td>
-			//<td></td>
-		//</tr>
+		console.log(this.store.cartItems);
+        let tbody = this.root.querySelector('tbody');
+        // using innerHTML to render a list of table row item under tbody
+        tbody.innerHTML = this.renderListAsHTML(this.store.cartItems)
+        let deleteButtons = this.root.querySelectorAll('.delete-button');
+        let fakeList = [{name: 'test', price: 9.99}, {name: 'test2', price: 9.59}];
+        for (var i = 0; i < deleteButtons.length; i ++) {
+        	let deleteBtn = deleteButtons[i];
+        	let scopedFakeList = fakeList;
+			deleteBtn.addEventListener('click', () => {
+        		debugger;
+        		// splice takes two arguments first being the index, second being
+        		// the number of items removing
+        		scopedFakeList.splice(0, 1);
+				alert('You are deleting' + deleteBtn);
+			});
+        }
     }
+	
+	/*
+	 * Input is a list of cart items (that were added through checkout button)
+	 *
+	 * Output is a String (which is HTML itself)
+	 */
+    renderListAsHTML(list) {
+		// replace with the for loop
+		//let result = '<tr><td>Name</td><td>Price</td><td><button class="delete-button" data-id="0">Delete</button></td></tr>';
+		//return result;
+		
+		list.forEach(function(item){
+			var result = "";
+			result = '<tr><td>item.name</td><td>item.price</td><td><button class="delete-button" data-id="0">Delete</button></td></tr>';
```

Comment:
This doesn't render correct name nor the price into the result string.

Please review the template string on how to pass variables into the template string. In example, the variable need to be surrounded with "${" and "}"